eating_cookies/eating_cookies.py

```diff
@@ -3,14 +3,58 @@
 import sys
 
 # The cache parameter is here for if you want to implement
-# a solution that is more efficient than the naive 
+# a solution that is more efficient than the naive
 # recursive solution
 def eating_cookies(n, cache=None):
-  pass
+    # Initialize the cache if it wasn't passed it
+    if cache == None:
+        cache = [0] * (n + 1)
+
+    if n <= 1:
+        cache[n] = 1
+        return 1
+    elif n == 2:
+        cache[n] = 2
+        return 2
+    elif n == 3:
+        cache[n] = 4
+        return 4
+    elif cache[n] != 0:
+        # print(n, cache)
+        return cache[n]
+    else:
+        cache[n] = (
+            eating_cookies(n - 1, cache)
+            + eating_cookies(n - 2, cache)
+            + eating_cookies(n - 3, cache)
+        )
+        return cache[n]
```

Comment:
Super cool. You're already ahead of the game, being able to implement the cache. We'll be using this technique again in the Computer Architecture week, so it will come in handy!